### PR TITLE
feat(formInput): adding 'input-button-prefix' and 'input-button-suffix' as input addons

### PR DESCRIPTION
**Pull request by rpocklin**
Will rebase tonight.

### CODE REVIEW

**Comment by rpocklin** [comment]
Rebased.

**Comment by uglow** [comment]
Sorry for the delay @rpocklin, I'll get to this next week (11th July)

File: src/modules/ngFormLib/controls/common/FormControlService.js [hunk 1]
```diff
@@ -185,19 +185,31 @@ mod.provider('formControlService', function() {
...
+        if (attr.inputPrefix || attr.inputSuffix || attr.inputButtonPrefix || attr.inputButtonSuffix) {
+          inputElem.wrap('<div class="input-group">');//inputElem.parent(); // This should be the 'control-row' element//wrap('<div class="input-group">');
+          let wrapper = inputElem.parent();
+
+          Object.keys(inputGroupMapping).forEach((inputGroupName) => {
+
+            if (attr[inputGroupName]) {
+              const inputGroupConfig = inputGroupMapping[inputGroupName];
+              const inputGroupContent = `<span class="${inputGroupConfig.className}">${attr[inputGroupName]}</span>`;
+              inputGroupConfig.prepend ? wrapper.prepend(inputGroupContent) : wrapper.append(inputGroupContent);
```
Comment (uglow) [inline src/modules/ngFormLib/controls/common/FormControlService.js]: If prepend was changed from a boolean property to a string property, you could write this instead:
```
...
inputButtonSuffix: {className: '...', attachFn: 'prepend'}
...
wrapper[inputGroupConfig.attachFn](inputGroupContent);
```

File: src/modules/ngFormLib/controls/common/FormControlService.js [hunk 2]
```diff
@@ -185,19 +185,31 @@ mod.provider('formControlService', function() {
...
+          inputSuffix: {className: 'input-group-addon', prepend: false},
+          inputButtonPrefix: {className: 'input-group-btn', prepend: true},
+          inputButtonSuffix: {className: 'input-group-btn', prepend: false}
+        };
+
+        if (attr.inputPrefix || attr.inputSuffix || attr.inputButtonPrefix || attr.inputButtonSuffix) {
+          inputElem.wrap('<div class="input-group">');//inputElem.parent(); // This should be the 'control-row' element//wrap('<div class="input-group">');
+          let wrapper = inputElem.parent();
+
+          Object.keys(inputGroupMapping).forEach((inputGroupName) => {
```
Comment (uglow) [inline src/modules/ngFormLib/controls/common/FormControlService.js]: I'm a bit concerned about Object.keys(), as IIRC PhantomJS doesn't understand it and you need to use a Babel transform to make it work for PhantomJS. But the tests have passed, so I'm surprised.

**Comment by uglow** [comment]
Some things need to be added: tests, documentation, purpose of the change. I've pulled your branch into my local branch and will continue working on it.

From looking at the bootstrap docs (http://getbootstrap.com/components/#input-groups-buttons) you need to render a <button> element for the styling to be correct. Is this what is needed?

File: src/modules/ngFormLib/controls/common/FormControlService.js (hunk 1 above)
Comment (rpocklin) [inline src/modules/ngFormLib/controls/common/FormControlService.js]: Sure.

File: src/modules/ngFormLib/controls/formInput/FormInput.js [hunk 3]
```diff
@@ -30,8 +30,8 @@ mod.directive('formInput', ['formControlService', function(formControlService) {
       formControlService.addLabelText(labelElem, tAttr.label);
       addPlaceholder(inputElem, formControlService.translate(tAttr.placeholder)); // Do this to be API-compatible with the form-select control. ff-placeholder is still supported. Use one or the other.
 
-      // If the user wants to use 'input-addon-prefix' or 'input-addon-suffix', change the DOM
-      var hasInputGroup = formControlService.addInputGroup(inputElem, tAttr.inputPrefix, tAttr.inputSuffix);
+      // If the user wants to use 'input-addon-*' or 'input-addon-*', change the DOM
```
Comment (rpocklin) [inline src/modules/ngFormLib/controls/formInput/FormInput.js]: not sure this comment makes sense to have these repeated (* covers both)

**Comment by uglow** [comment]
Closing to PR #15

**Pull request closed without merging**